### ISSUE TITLE
Decide what to do with CFlow

### ISSUE THREAD

**Issue by Robyt3** [teeworlds/teeworlds#2437]
[`CFlow`](https://github.com/teeworlds/teeworlds/blob/master/src/game/client/components/flow.cpp) is effectively unused since 0.5. Only `Add` and `Update` are called. Neither `Get` nor `DbgRender` are used, making these calculations effectively useless. The related config var is commented out
`//MACRO_CONFIG_INT(ClFlow, cl_flow, 0, 0, 1, CFGFLAG_CLIENT|CFGFLAG_SAVE, "")`. Even then only being
> a test feature in the past - @oy 2011

I can see the use this could have, but then we should actually make use of it and not have it sitting around in an incomplete state.

Or it should just be removed, because having been unused all those years might be an indication that there isn't a good use for it, or the effects system on it's own works well enough.

**Comment by AssassinTee** [comment]
Can you explain what its purpose is?
For me it looks like an abandoned class.

**Comment by Robyt3** [comment]
Yes, it's mostly abandoned at this point. It seems that the idea was to try and simulate particles or effects using a 2D field of velocity vectors. Explosion effects call `Add` to create flow, which decreases over time with `Update`. However the values are never used anywhere, e.g. to influence other particles, so there is no use to it.

**Comment by oy** [comment]
There used to be a video about smoke reacting when a player moves on by, can't find it. Think that was related to this.

**Comment by heinrich5991** [comment]
> There used to be a video about smoke reacting when a player moves on by, can't find it.

Perhaps these?

https://heinrich5991.de/teeworlds/video/teewars8.ogm
https://heinrich5991.de/teeworlds/video/teewars9.ogm

**Comment by oy** [comment]
@heinrich5991 yeah, nice. These are the ones 👍

**Comment by AssassinTee** [comment]
looks cool, why did you not use this?

**Comment by Dune-jr** [comment]
Probably clutters the screen too much? Given that it's a very old video, I'm not sure many people know :)
